include/dynd/func/arithmetic.hpp

```diff
@@ -14,11 +14,11 @@
 namespace dynd {
 namespace nd {
 
-  template <typename F, template <type_id_t...> class K, int N>
+  template <typename F, template <type_id_t...> class K, int N, typename TypeIDs>
   struct arithmetic_operator;
 
```

Comment:
Adding the TypeIDs as a template parameter is a good step for sure. A trivial comment is that I typically have called such a typename TypeIDSequence throughout DyND, so we should harmonize that.

A less trivial comment is I'd like to hear your opinions on this arithmetic_operator template. This template probably should be a more general thing, as we have a similar comparision_operator template and so on. I'm not sure if arithmetic_operator was a good idea or not, it was a bit experimental when I wrote it.